app/src/main/java/com/github/featuredetectandroid/utils/PhotoAnalyzer.kt

```diff
@@ -0,0 +1,50 @@
+package com.github.featuredetectandroid.utils
+
+import android.util.Log
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import com.github.featuredetectandroid.ui.GrayscaleViewModel
+import java.nio.ByteBuffer
+
+private const val TAG = "PhotoAnalyzer"
+private const val ROTATION_STEP = 90
+
+class PhotoAnalyzer(private val imageViewModel: GrayscaleViewModel) : ImageAnalysis.Analyzer {
+    private fun ByteBuffer.toByteArray(): ByteArray {
+        rewind()
+        return ByteArray(remaining()).also { get(it) }
+    }
+
+    private fun rotateClockwiseOnRotationStep(
+        grayscaleByteArray: ByteArray,
+        width: Int,
+        height: Int
+    ) = ByteArray(width * height).also { array ->
+        for (i in 0 until width) {
+            for (j in 0 until height) {
+                array[i * height + j] = grayscaleByteArray[i + (height - 1 - j) * width]
+            }
+        }
+    }
+
+    // Rotation of incorrectly oriented images is implemented here.
+    @Suppress("ForbiddenComment")
```

Comment:
> Maybe it would be better to put this comment where the rotation happens (inside the function) and get rid of this supression?

"ForbiddenComment" is comment with "TODO".

Reply:
fixed